src/index.js

```diff
@@ -0,0 +1,108 @@
+import './styles.scss';
+import celsiusToFahrenheit from './functions';
+import showError from './dom';
+
+// get the elements
+const weatherIcon = document.querySelector('.weather-icon');
+const weatherDescription = document.querySelector('.description');
+const tempCelsius = document.querySelector('.temp-celsius');
+const tempMin = document.getElementById('temMin');
+const tempMax = document.getElementById('temMax');
+const location = document.querySelector('.location');
+const todayDate = document.getElementById('todayDate');
+const inputCity = document.getElementById('address-input');
+
+// app data
+const weather = {};
+weather.temperature = {
+  unit: false,
+};
+
+const KELVIN = 273;
+const KEY = '15b72f8181c849f71bb8b90b88730574';
+
+const displayWeather = () => {
+  weatherIcon.innerHTML = `<img src="http://openweathermap.org/img/wn/${weather.iconId}@2x.png">`;
+  tempCelsius.innerHTML = `${weather.temp}°C`;
+  tempMin.innerHTML = `${weather.temp_min}|`;
+  tempMax.innerHTML = `${weather.temp_max}`;
+  weatherDescription.innerHTML = weather.description;
+  location.innerHTML = `${weather.city},${weather.country}`;
+  todayDate.innerHTML = `${new Date().toDateString()}`;
+};
+
+const getWeather = async (latitude, longitude) => {
+  const api = `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${KEY}`;
+  try {
+    const fetchRequest = await fetch(api);
+    const data = await fetchRequest.json();
+    weather.temp = Math.floor(data.main.temp - KELVIN);
+    weather.temp_min = Math.floor(data.main.temp_min - KELVIN);
+    weather.temp_max = Math.floor(data.main.temp_max - KELVIN);
+    weather.description = data.weather[0].description;
+    weather.iconId = data.weather[0].icon;
+    weather.city = data.name;
+    weather.country = data.sys.country;
+    displayWeather();
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.log(err);
```

Comment:
- [ ] No `console.log` commited to repo. It kills performance.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/console_log_requirement.png)

We can replace the error handling by displaying the information in a notification box.